aws/signer/v4/presign_middleware.go

```diff
@@ -30,33 +31,42 @@ type PresignedHTTPRequest struct {
 	SignedHeader http.Header
 }
 
-// PresignHTTPRequestMiddleware provides the Finalize middleware for creating a
+// PresignHTTPRequestMiddlewareOptions is the options for the PresignHTTPRequest middleware.
+type PresignHTTPRequestMiddlewareOptions struct {
+	CredentialsProvider aws.CredentialsProvider
+	Presigner           HTTPPresigner
+	LogSigning          bool
+}
+
+// PresignHTTPRequest provides the Finalize middleware for creating a
 // presigned URL for an HTTP request.
 //
 // Will short circuit the middleware stack and not forward onto the next
 // Finalize handler.
-type PresignHTTPRequestMiddleware struct {
+type PresignHTTPRequest struct {
```

Comment:
Similar to SignHTTPRequest. Since these types are in same package as `Signer` and `Presigner` they might get confusing without the `Middleware` suffix on their name. Also since their constructors have that on their name.